crates/typstyle-core/src/lib.rs

```diff
@@ -27,63 +27,61 @@ impl std::fmt::Display for FormatError {
 }
 
 /// Entry point for pretty printing a typst document.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Default)]
 pub struct Typstyle {
-    source: Source,
-    config: PrinterConfig,
+    config: Config,
 }
 
 impl Typstyle {
-    /// Create a new Typstyle instance from a string.
-    /// # Example
-    /// ```rust
-    /// use typstyle_core::{PrinterConfig, Typstyle};
-    /// let content = "#{1+1}";
-    /// let res = Typstyle::new_with_content(content.to_string(), PrinterConfig {
-    ///     max_width: 80,
-    ///     ..Default::default()
-    /// }).pretty_print();
-    /// ```
-    pub fn new_with_content(content: impl Into<String>, config: PrinterConfig) -> Self {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn config(mut self, config: Config) -> Self {
+        self.config = config;
+        self
+    }
+
+    pub fn config_width(mut self, max_width: usize) -> Self {
```

Comment:
maybe user should take a `&mut` of config and edit it directly? I'm not sure but this interface doesn't look good to me. Or we may forbid user from editing config after typstyle is constructed.

Reply:
cc @Myriad-Dreamin

Reply:
Reconfiguration can happen when hot reloading lsp configurations but I'm not sure whether it is good to expose a `&mut config` directly because you may derive and store some value from config internally and it will cause inconsistency to change config by the exposed mut.

Reply:
Currently we don't have extra state in config. If that's needed, we might need to change the struct hierarchy here. Maybe we can use a simple design at this moment and revisit it when we need more functions.

BTW I just realized that currently we will create a new typstyle instance every time when we format a file. So technically it is "reconfigured" every single time.